services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/guideline.rb

```
@@ -34,11 +34,21 @@ class Guideline < ApplicationRecord
         scope :visible, -> { where(visible: true) }
         scope :archived, -> { where(visible: false) }
 
+        after_create :duplicate_for_dataset_parent
+
         def self.assigned_status_column = :curriculum_assigned_status
 
         def to_s = text
 
         def copy_attributes = { curriculum_assigned_status:, notes:, text:, visible: }
+
+        def duplicate_for_dataset_parent
```

You can collapse these three lines into one; also, I would use `create!` otherwise this will fail silently:

`Guideline.create!(copy_attributes.merge(dataset_id: dataset.parent_id)`